Loop/Managers/DeviceDataManager.swift

```diff
@@ -1210,6 +1211,67 @@ extension DeviceDataManager {
                 case .cancelTemporaryOverride:
                     log.default("Canceling temporary override from remote command")
                     loopManager.settings.scheduleOverride = nil
+                case .bolusEntry(let bolusAmount):
+                    log.default("Enacting remote bolus entry: %{public}@", String(describing: bolusAmount))
+                    
+                    //Remote bolus requires validation from its remote source
+                    guard remoteDataServicesManager.validatePushNotificationSource(notification) else {
+                        NotificationManager.sendRemoteBolusFailureNotification(for: RemoteCommandError.invalidOTP, amount: bolusAmount)
+                        log.info("Could not validate notification: %{public}@", String(describing: notification))
+                        return
+                    }
+                    
+                    guard let maxBolusAmount = loopManager.settings.maximumBolus else {
+                        NotificationManager.sendRemoteBolusFailureNotification(for: RemoteCommandError.missingMaxBolus, amount: bolusAmount)
+                        log.default("No max bolus detected. Aborting...")
+                        return
+                    }
+                    
+                    guard bolusAmount.isLessThanOrEqualTo(maxBolusAmount) else {
+                        NotificationManager.sendRemoteBolusFailureNotification(for: RemoteCommandError.exceedsMaxBolus, amount: bolusAmount)
+                        log.default("Remote bolus higher than maximum. Aborting...")
+                        return
+                    }
+                    
```

Comment:
When enacting the bolus here, we may get a specialized error from the enactBolus API which will also be presented in a local notification.... otherwise we show a local notification with a success message.